Storefront blog page

Ziuziun-Pavel opened the pull request.

Commit:
build_separate_page_for_post

Commit:
get_post_by_id

Commit:
add_styles

aleh-ivanov-innowise commented on Service/CategoriesProvider.php:
```diff
@@ -0,0 +1,28 @@
+
+    public function getCategories(): Collection
+    {
+        $collection = $this->getCollection();
```
Почему-бы не использовать репозиторий?

aleh-ivanov-innowise commented on Service/PostsProvider.php:
```diff
@@ -0,0 +1,30 @@
+
+    public function getPosts(int $limit, int $currentPage): Collection
+    {
+        $collection = $this->getCollection();
```
Почему-бы не использовать репозиторий?

aleh-ivanov-innowise commented on ViewModel/PostViewModel.php:
```diff
@@ -0,0 +1,35 @@
+class PostViewModel implements ArgumentInterface
+{
+    private $collection;
+    private $postRepositoryInterface;
```
В имени переменных обычно не указывается `Interface`

aleh-ivanov-innowise commented on Controller/Index/Index.php:
```diff
@@ -0,0 +1,24 @@
+use Magento\Framework\Controller\ResultInterface;
+use Magento\Framework\View\Result\PageFactory;
+
+class Index implements HttpGetActionInterface
```
Какой формат url-а этой страницы?

aleh-ivanov-innowise replied:
Полагаю, нужно добавить кастомный роутер, чтобы сделать url-ы блог постов и категорий seo friendly (в постах и категориях хранится url-key)

aleh-ivanov-innowise commented on view/frontend/templates/nav_bar.phtml:
```diff
@@ -0,0 +1,34 @@
+        data-mage-init='{"menu":{"responsive":true, "expanded":true, "delay": 200, "position":{"my":"left top","at":"left+10 top+30"}}}'>
+        <li class="level0 level-top ui-menu-item">
+            <a class="level-top ui-corner-all" id="ui-id-8" tabindex="-1" role="menuitem"
+               href="https://<?php echo $_SERVER['HTTP_HOST'] ?>/blog">
```
Так в Мадженто не делается, нужно генерировать url-ы через UrlBuilder

aleh-ivanov-innowise replied:
Также любой вывод в шаблоны должен экранироваться, https://developer.adobe.com/commerce/php/development/security/cross-site-scripting/#phtml-templates

aleh-ivanov-innowise commented on view/frontend/web/styles/blog.css:
```diff
@@ -0,0 +1,57 @@
+ul {
```
Попробуй сделать это через less

Commit:
change-some-comments-and-start-working-with-CustomPager

Commit:
 This is the 1st commit message:

change-some-comments-and-start-working-with-CustomPager

 The commit message #2 will be skipped:

 get_post_by_id

 The commit message #3 will be skipped:

 add_styles

Commit:
change-css-file-to-less

Ziuziun-Pavel committed:
add customPager working with repository

Ziuziun-Pavel committed:
start building of custom router and get posts by url_key

Ziuziun-Pavel committed:
implement custom router

Ziuziun-Pavel committed:
fixed bugs connected with router

Ziuziun-Pavel committed:
fixed routes using UrlBuilder and escape the outputs in the templates

Ziuziun-Pavel committed:
add category page with related posts, fixed routes to make them seo-friendly

aleh-ivanov-innowise commented on Block/Html/CustomPager.php:
```diff
@@ -0,0 +1,67 @@
+     */
+    public function toHtml()
+    {
+        $this->setTemplate('Innowise_Blog::custom_pager.phtml');
```
Зачем здесь сетать template, если он прописан в layout?

aleh-ivanov-innowise commented on Block/Html/CustomPager.php:
```diff
@@ -0,0 +1,67 @@
+    /**
+     * Set the collection for the pager.
+     *
+     * @param \Innowise\Blog\Api\Data\PostSearchResultsInterface $collection
```
Везде по коду: если это уже не коллекция, а search results, то для избежания путаницы стоит переименовать соотв. переменные и методы. Например, вместо `setCollection($collection)` - `setPostSearchResults(PostSearchResultsInterface $searchResults): CustomPager `

Commit:
fixed bugs

Commit:
fixed bugs

Commit:
fixed bugs
